#! /usr/bin/env python
PACKAGE="tuw_keyboard"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
                                                                        
gen.add("rate",    double_t,    0, "[Hz] command publish rate",  10,  0, 50)

gen.add("init_v", double_t, 0, "[m/s] initial translation velocity",   0.2, 0,   5)
gen.add("init_w", double_t, 0, "[rad/s] initial rotation velocity",    0.2, 0,   5)

gen.add("max_v", double_t, 0, "[m/s] max translation velocity",    2, 0,   5)
gen.add("max_w", double_t, 0, "[rad/s] max rotation velocity",    1.57, 0,   5)

gen.add("steps_v",    double_t,    0, "[m/s] step size translation velocity", 0.1,  -1, 1)
gen.add("steps_w",    double_t,    0, "[rad/s] step size rotation velocity", 0.1,  -1, 1)

gen.add("wheel_radius", double_t, 0, "needed for IWS control: wheel radius",    0.075, 0,   1)
gen.add("wheel_displacement", double_t, 0, "needed for IWS control: distance between wheels",    0.449, 0,   2)

exit(gen.generate(PACKAGE, "tuw_keyboard", "KeyboardControl"))
